internal/pkg/cli/pipeline_list.go

```diff
@@ -27,43 +32,78 @@ import (
 const (
 	pipelineListAppNamePrompt = "Which application are the pipelines in?"
 	pipelineListAppNameHelper = "An application is a collection of related services."
+
+	pipelineListTimeout = 10 * time.Second
 )
 
 type listPipelineVars struct {
-	appName          string
-	shouldOutputJSON bool
+	appName                  string
+	shouldOutputJSON         bool
+	shouldShowLocalPipelines bool
 }
 
 type listPipelineOpts struct {
 	listPipelineVars
-	codepipeline   pipelineGetter
-	pipelineLister deployedPipelineLister
 	prompt         prompter
 	sel            configSelector
 	store          store
 	w              io.Writer
+	workspace      wsPipelineGetter
+	pipelineLister deployedPipelineLister
+
+	newDescriber newPipelineDescriberFunc
+
+	wsAppName string
 }
 
+type newPipelineDescriberFunc func(pipeline deploy.Pipeline) (describer, error)
+
 func newListPipelinesOpts(vars listPipelineVars) (*listPipelineOpts, error) {
+	ws, err := workspace.New()
+	if err != nil {
+		return nil, err
+	}
+
 	defaultSession, err := sessions.ImmutableProvider(sessions.UserAgentExtras("pipeline ls")).Default()
 	if err != nil {
 		return nil, fmt.Errorf("default session: %w", err)
 	}
+
+	var wsAppName string
+	if vars.shouldShowLocalPipelines {
+		wsAppName = tryReadingAppName()
+		if vars.appName == "" {
+			vars.appName = wsAppName
+		}
```

Comment:
Line 155 has: `cmd.Flags().StringVarP(&vars.appName, appFlag, appFlagShort, tryReadingAppName(), appFlagDescription)`

Which gives `appName` a default value of the workspace app, which gets overridden if `--app` is used. This block might be doing the same thing?

Reply:
good catch! 😊